python/vTestInterface.py

```diff
@@ -560,7 +580,99 @@ def validateClicastCommand (command, mode):
             raise UsageError()
         
 
-def processCommand (mode, clicast, pathToUse, testString="") -> dict:
+def updatedEnvironment (enviroPath, jsonOptions):
+    """
+    pathToUse is the full path to the environment directory
+    jsonOptions has the new values of ENVIRO.* commands for the enviro script
+    e.g.  ENVIRO.COVERAGE_TYPE: Statement
+
+    We overwrite any matching ENVIRO commands with the new values befoe rebuild
+    """
+
+    tempEnviroScript  = "rebuild.env"
+    tempTestScript = "rebuild.tst"
+
+    with cd(os.path.dirname(enviroPath)):
+
+        # first we generate a .env and .tst for the existing environment
+        # we do this using a clicast script
+        enviroName = os.path.basename(enviroPath)
+        with open(commandFileName, "w") as commandFile:
+            commandFile.write(f"-e{enviroName} enviro script create {tempEnviroScript}\n")
+            commandFile.write(f"-e{enviroName} test script create {tempTestScript}\n")
+        exitCode, stdOutput = runClicastScript(commandFileName, True)
+
+
+        # Read the enviro script into a list of strings
+        with open (tempEnviroScript, "r") as enviroFile:
+            enviroLines = enviroFile.readlines()
+
+        # Re-write the enviro script replacing the value of commands
+        # that exist in the jsonOptions
+        with open (tempEnviroScript, "w") as enviroFile:
+            for line in enviroLines:
+                whatToWrite = line
+                if line.startswith ("ENVIRO.END"):
+                    # if we have some un-used options then 
+                    # write these before the ENVIRO.END
+                    for key, value in jsonOptions.items():
+                        enviroFile.write (f"{key}: {value}\n")
+
+                elif line.startswith ("ENVIRO.") and ":" in line:
+                    # for all other commands, see if the command matches
+                    # a command from the jsonOptions dict
+                    enviroCommand, enviroValue = line.split (":",1)
+                    enviroCommand = enviroCommand.strip()
+                    enviroValue = enviroValue.strip()
+                    # if so replace the existing value ...
+                    if enviroCommand in jsonOptions:
+                        whatToWrite =  (f"{enviroCommand}: {jsonOptions[enviroCommand]}\n")
+                        jsonOptions.pop (enviroCommand)
+
+                # write the orignal or updated line
+                enviroFile.write (whatToWrite)
+
+        # Finally delelete and re-build the environment using the updated script
+        # and load the existing tests -> which duplicates what enviro rebuild does.
+        with open(commandFileName, "w") as commandFile:
+            # Improvement needed: vcast bug: 100924
+            shutil.rmtree (enviroName)
+            #commandFile.write(f"-e{enviroName} enviro delete\n")
```

Comment:
I guess the `delete` command doesn't work properly, so we are using `shutil.rmtree`